### PR TITLE
Add support for supportedCardBrands to HostedFieldsField in braintree-web hosted-fields.d.ts

### PR DESCRIPTION
**Pull request by MatthewWierzbicki**
- Adding support for supportedCardBrands to HostedFieldsField in braintree-web hosted-fields.d.ts
- [Braintree Hosted Fields Docs](https://braintree.github.io/braintree-web/current/module-braintree-web_hosted-fields.html#~field)

Valid card brand ids are:

- visa
- mastercard
- american-express
- diners-club
- discover
- jcb
- union-pay
- maestro
- elo
- mir
- hiper
- hipercard

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:
If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## braintree-web ([<kbd>unpkg</kbd>](https://unpkg.com/browse/braintree-web@latest/))
was missing the following properties:
1. masterpass
1. paymentRequest
1. visaCheckout
1. sepa
1. preferredPaymentMethods
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c9f6ffce93863148b2d98af789fc42b5572770df
</p>

File: types/braintree-web/paypal-checkout.d.ts [hunk 1]
```diff
@@ -43,15 +43,11 @@ export interface PayPalCheckoutLoadPayPalSDKOptions {
      * wish to use a different intent when calling createPayment, make sure it matches here. If sale
      * is used, it will be converted to capture for the PayPal SDK. If the vault: true param is used,
      * no default intent will be passed.
-     *
-     * @default 'authorize'
```
Comment (MatthewWierzbicki) [inline types/braintree-web/paypal-checkout.d.ts:47]: Had to remove for tests to pass, deprecated

Comment (daelmaak) [reply types/braintree-web/paypal-checkout.d.ts:47]: Why did you have to remove it? Other people didn't have this problem and it should definitely stay in the JS doc.

**Review by daelmaak — COMMENTED** [review]
Thanks for the addition, but we have to clear up the `@default`

**Comment by MatthewWierzbicki** [comment]
> Thanks for the addition, but we have to clear up the `@default`

@daelmaak No problem whatsoever 🙏. Hmmmm, I'm not really sure how to fix this in that case, we can ignore the error if that is preferred but I don't think it's a good idea to change this lint config globally. 

Here is a link to the previous ci/cd pipeline: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/66860/checks?check_run_id=17193442225

```
ERROR: 47:9  no-redundant-jsdoc  JSDoc tag '@default' is redundant in TypeScript code.
ERROR: 54:9  no-redundant-jsdoc  JSDoc tag '@default' is redundant in TypeScript code.
ERROR: 68:9  no-redundant-jsdoc  JSDoc tag '@default' is redundant in TypeScript code.
```

File: types/braintree-web/paypal-checkout.d.ts (hunk 1 above)
Comment (MatthewWierzbicki) [reply types/braintree-web/paypal-checkout.d.ts]: Left a reply below 🙏

**Comment by daelmaak** [comment]
> JSDoc tag '@default' is redundant in TypeScript code.

There is no easy fix apart from disabling the TSLint rule in question on those lines, since on one hand it doesn't make sense for interface properties but on the other we don't want change the project's config atm. Could you please do the change?

**Comment by MatthewWierzbicki** [comment]
> > JSDoc tag '@default' is redundant in TypeScript code.
> 
> There is no easy fix apart from disabling the TSLint rule in question on those lines, since on one hand it doesn't make sense for interface properties but on the other we don't want change the project's config atm. Could you please do the change?

Kk, disabled the rule for those lines only, not for the whole package 👍

**Review by daelmaak — APPROVED** [review]
Thanks.

**Comment by MatthewWierzbicki** [comment]
Ready to merge

**Pull request merged**